src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java

```diff
@@ -5818,5 +5820,59 @@ public String getScope()
         return "public";
     }
 
+    @Test 
+    public void testRetrieveNodeDefinition() throws Exception
+    {
+        setRequestContext(networkOne.getId(), user1, null);
+        
+        String node1 = "nodeSample" + RUNID + "_1";
+        String node1Type = TYPE_CM_CONTENT;
+        Map<String,Object> props = new HashMap<>();
+        props.put("cm:name", "my test node");
+        Node node = createNode(Nodes.PATH_MY, node1, node1Type, props);
+        String nodeId = node.getId();
+        
+        HttpResponse  response = getSingle(NodesEntityResource.class, nodeId, null, 200);
+        Node nodeResp = RestApiUtil.parseRestApiEntry(response.getJsonResponse(), Node.class);
+        assertNull("Definition should not be retrieved unless included!", nodeResp.getDefinition());
+
+        Map params = new HashMap<>();
+        params.put("include", "definition");
+        response = getSingle(NodesEntityResource.class, nodeId, params, 200);
+        nodeResp = RestApiUtil.parseRestApiEntry(response.getJsonResponse(), Node.class);
+        NodeDefinition nodeDefinition = nodeResp.getDefinition();
+        assertNotNull(nodeDefinition);
+        checkDefinitionProperties(nodeDefinition.getProperties());
+    }
+    
+    private void checkDefinitionProperties(List<NodeDefinitionProperty> properties)
+    {
+        assertNotNull(properties);
+        NodeDefinitionProperty testProperty = properties.stream().
+                filter(property -> 
+                        property.getId().equals("cm:name"))
+                .findFirst()
+                .get();
+        assertNotNull(testProperty);
+        assertNotNull(testProperty.getConstraints());
+
+        shouldNotContainSystemProperties(properties);
+        shouldContainParentProperties(properties);
```

Comment:
Please also add some tests to check the individual (JSON) properties of a (node) properties object i.e. dataType, isProtected etc. Picking specific well know properties from the default content model as you've done is fine.

Would also be good to check an individual constraint object too.

Reply:
I added only some basic test to check value of properties and constraints. Picking types from the default content model cover only few attributs, I will try to use a more complex content model in future iteration/refactoring for this test